src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -156,27 +154,32 @@ public virtual void Abort()
                             {
                                 if (NegotiationProtocol.TryParseMessage(buffer, out var negotiationMessage, out consumed, out examined))
                                 {
-                                    var protocol = protocolResolver.GetProtocol(negotiationMessage.Protocol, supportedProtocols, this);
+                                    Protocol = protocolResolver.GetProtocol(negotiationMessage.Protocol, supportedProtocols, this);
 
                                     var transportCapabilities = Features.Get<IConnectionTransportFeature>()?.TransportCapabilities
                                         ?? throw new InvalidOperationException("Unable to read transport capabilities.");
 
-                                    var dataEncoder = (protocol.Type == ProtocolType.Binary && (transportCapabilities & TransferMode.Binary) == 0)
-                                        ? (IDataEncoder)Base64Encoder
-                                        : PassThroughEncoder;
+                                    var requiredTransferMode = (Protocol.Type == ProtocolType.Binary) ?
```

Comment:
nit: I prefer the `?` and `:` to be on the start of new lines like the removed code here was, but I'm not sure what y'alls standard is

Reply:
I don't know that we have a standard, but this seems as good as any :)